### DISCUSSION TITLE
Adding refracted/transmitted components to Sionna

### DISCUSSION THREAD

**Discussion by cplonski20** [NVlabs/sionna#441] [Q&A]
How difficult would it be to modify the current code to track refracted/transmitted components in Sionna? Could someone knowledgeable on the code base direct me towards the relevant files and give me an idea for how much work this would take?

**Comment by faycalaa** [comment]
Hi,

You would need to implement transmission/refraction for the compute of both channel impulse responses (CIRs) ([solver_paths.py](https://github.com/NVlabs/sionna/blob/main/sionna/rt/solver_paths.py)) and coverage maps ([solver_cm.py](https://github.com/NVlabs/sionna/blob/main/sionna/rt/solver_cm.py)). Note that the compute of CIR and coverage map is implemented by two very different algorithms, hence the separate files.

**Comment by JackYifan** [comment]
@faycalaa  May I ask if you have now realized the calculation of the refractive field? I have a need to calculate the refractive field, too.

**Reply by faycalaa** [reply]
Hi @JackYifan ,

There is currently no support for refraction.

**Reply by JackYifan** [reply]
@faycalaa Will sionna add this feature in the future?
